feature-configs/cn-ran-overlays/ran-profile-gcp/pre_deploy.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+export YQ_DIR="${YQ_DIR:-/tmp}"
+export YQ_BIN=$YQ_DIR/yq
+export YQ_VERSION=v4.18.1
+
+set -e
+
+echo "****** ran-profile-gcp pre_deploy.sh ******"
+echo "FEATURES_ENVIRONMENT: ${FEATURES_ENVIRONMENT}"
+
+echo "PRINT current location: $(pwd)"
+echo "PRINT script: $0"
+obsolute_feature_dir=$(dirname $(realpath "$0"))
+feature=$(basename ${obsolute_feature_dir})
+ztp_generator_cli=ztp/resource-generator/entrypoints/generator
+source_pgt_dir=$(dirname ${obsolute_feature_dir})/ran-profile/policygentemplates
+
+# set variables for ztp generator
+export policyGenPath=${obsolute_feature_dir}/policyGen
+
+echo "PRINT feature dir: ${obsolute_feature_dir}"
+echo "PRINT feature: ${feature}"
+
+# prepare resources required by ztp generator
+mkdir -p ${policyGenPath}
+cp -r ztp/source-crs ${policyGenPath}
+go build -o ${policyGenPath}/PolicyGenTemplate ./ztp/policygenerator
+
+# call ztp generator script to generate the non-policy wrapped cluster config CRs based on the du profile defined in PolicyGenTemplates
+source_config_CRs="${ztp_generator_cli} config -N ${source_pgt_dir}/common-ranGen.yaml ${obsolute_feature_dir}/cluster-config"
+if ! $source_config_CRs; then
+  echo "Failed to generate the source ran profile CRs"
+  exit 1
+fi
+
+gcp_config_CRs="${ztp_generator_cli} config -N ${obsolute_feature_dir}/policygentemplates/common-ranGen.yaml ${obsolute_feature_dir}/cluster-config"
+if ! $gcp_config_CRs; then
+  echo "Failed to generate the ran profile CRs for GCP"
+  exit 1
+fi
+
+# Auto-create kustomization yaml
+if [[ ! -d ${obsolute_feature_dir}/cluster-config ]]; then
+  echo "cluster-config doesn't exist"
+  exit 1
+fi
+
+cd ${obsolute_feature_dir}
+cat > kustomization.yaml << EOF
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+EOF
+
+find cluster-config/ -type f | while read config_yaml; do
```

Comment:
Maybe we should add `-name='*.yaml'` to the find command to filter out potential other files like READMEs or editor swapfiles?